build/dist

```diff
@@ -248,7 +248,18 @@ echo -e "\nBuilding with..."
 # shellcheck disable=SC2145
 echo -e "\$ ${BUILD_COMMAND[@]}\n"
 
+# shellcheck disable=SC2050
+if [ "$SCALA_VERSION" = "2.12" ]; then
+  EXTRA_BUILD_COMMAND=("$MVN" install $MVN_DIST_OPT $@  -Pscala-2.13 -pl :kyuubi-spark-sql-engine_2.13 )
+else
+  EXTRA_BUILD_COMMAND=("$MVN" install $MVN_DIST_OPT $@  -Pscala-2.12 -pl :kyuubi-spark-sql-engine_2.12 )
```

Comment:
Thanks @PorterZhang2021 contribution, there are some minor comments:

- Apache kyuubi does not have a profile of the scala-2.12, you can use the default profile for scala-2.12;
- `$@` is actually unnecessary;

You can also [create a kyuubi tarball](https://kyuubi.readthedocs.io/en/master/contributing/code/distribution.html), and use it to manual E2E test.

Reply:
`$@` should be persisted, it's user passed custom mvn args

Reply:
yea, but in the case of actually compiling only the spark engine jar, the additional parameters may not be necessary and may instead affect the compilation of the current module;

For example, if the user create a kyuubi tarball compiled by `./dev/dist -Pscala-2.13`, we need to compile the spark engine jar of `scala-2.12`. then the compile command for this block becomes 
```shell
"$MVN" install $MVN_DIST_OPT **-Pscala-2.13** ~~-Pscala-2.12~~ -pl: Kyuubi-spark-SQl-engine_2.2.12 -am?
```

BTW, kyuubi doesn't seem to have a profile for scala-2.12?

Reply:
Hmm... this definitely is an issue, but we can not skip all user-provided mvn args too. we need to rethink and make a followup

Reply:
> kyuubi doesn't seem to have a profile for scala-2.12?

yes, one more thing we need to verify, if both `scala-2.12` and `scala-2.13` exist, which one will take effect if both profiles are activated

Reply:
It is better not to rely on some small feature of maven to complicate things, if only the spark engine jar is compiled, what user-provided parameters might be useful?

Reply:
for example, override dep's version

Reply:
Oh, I see. then we need another way to solve this problem.